pkg/interfaces/host.go

```diff
@@ -8,10 +8,17 @@ import (
 
 type IHostService interface {
 	CreateHost(*domain.Host) (*domain.Host, error)
-	GetHostsByTenantID(tenantID string) ([]*domain.Host, error)
+	GetHostsByTenantIDAndUserID(tenantID string, userID string) ([]*domain.Host, error)
+	GetHostByID(ID string) (*domain.Host, error)
+	GetHostname(string) string
+	DeleteHostByID(ID string) (bool, error)
+	PatchHostByID(ID, domainName, ip, alias string, credential, rapporteur []byte) (*domain.Host, error)
```

Comment:
With the refactor I suggested, this PatchHostByID method would receive an `ID` and `domain.Host` parameter and return `(*domain.Host, error)`